### DISCUSSION TITLE
[Feature] GPS Prediction/Correction (using i.e. GeoClip)

### DISCUSSION THREAD

**Discussion by nielstron** [immich-app/immich#12584] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Many of my pictures were taken with an old phone/camera where I turned off GPS tracking or the GPS metadata was lost in the digitalization process. It would be really nice to restore this information using models like GeoCLIP (https://github.com/VicenteVivan/geo-clip), maybe also combine the guessed GPS location with the closeby-same-time information from https://github.com/immich-app/immich/discussions/1675.

I tested this model myself and often enough the prediction is pretty bad (but corresponding probabilities are rated very low too). Therefore it would be nice to add the suggestion to the GPS location picker and/or accept and override the existing data only if the certainty is very high.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jesjimher** [comment]
This would be very nice for all those old photos taken before smartphones. Another similar model:

https://huggingface.co/geolocal/StreetCLIP

**Comment by bo0tzz** [comment]
#9501

**Reply by nielstron** [reply]
I don't think this is a duplicate. Rather than predicting the GPS location and actually directly using it (for this the tool is definitely too bad) I suggest integrating it for suggestions when manually assigning a location via the location picker (here the precision of the tool does not matter, users can discard the suggestion if it does not make sense and otherwise it may be a huge help)
